### ISSUE TITLE
Noticing battery life on the dash button significantly shorter since blocking its internet access

### ISSUE THREAD

**Issue by OrangeFlavored** [Nekmo/amazon-dash#154]
Not sure if anyone else has noticed or brought this to your attention, but it seems like when you block the button from phoning home it spends a very long time flashing white and attempting to make a connection each time it is pressed, to the point that one of my buttons I activated a couple weeks ago has already drained 50% of its battery after being pressed twice a day.   Another one started with 54 or so and is now "dead" (it reports 13% or so but apparently that's low enough that it's triggering the auto-wipe and won't do anything useful anymore)  Seems to me that back in the old days before we worried about blocking them from phoning home I'd get several months out of a button before I had to worry about them running dry.  I'm not sure if anything can actually be done about this on your end, but bring it up just on the off chance maybe it can, like some kinda signal be sent to get it to stop trying to connect after the initial pulse goes out or something.

Also I did accidentally leave one button unblocked on my network for a little while and noticed it behaved how it used to (4 or so white flashes and then a quick red flash and shut off) without getting bricked or anything, so I'm kinda tempted to just try leaving one unblocked for a little while and see if maybe Amazon stopped bricking them by now and just returns a generic connection error from their side, since it seems like that'd be nicer on the battery.  Kind of suspect this might be the case, like why waste their money keeping update servers up for a device they have no intention of supporting anymore?  Guess there's not much harm in trying it with the one that's down to 39% or so since it'd probably be dead in another week or two anyway at this rate.  Firmware didn't get updated either (to bricking or non-bricking version) after a few presses like that, green light exploit still worked fine after all was done.  Anyone have any info on whether it's safe to just let them access the internet at this point?

Maybe blocking via DNS in my router rather than disabling internet access to the dash button's mac address would also trigger the shorter loop, maybe I'll try that later.  For now I guess I'll experiment with leaving one button unblocked and maybe report back if the thing actually gets bricked/updated in the near future.

**Comment by Nekmo** [comment]
Sometimes the battery of my buttons goes up and other times it goes down ... Maybe the percentage has never worked. Did you check the battery percentage before Amazon shutdown?

**Comment by OrangeFlavored** [comment]
I only checked it when first setting up the two buttons (sitting in a box from prime day some time ago and unused until now besides initial press) and then again when I noticed one of them wasn't working anymore, hadn't checked between then to see how it was moving though.

Well, either way I think the quick 3-4 white -> quick red -> off is preferable to the thing flashing white for 30 seconds or so.  Not sure if another method of blocking like DNS redirection would cause it to behave this way instead of the long white flashing, I feel like it's inevitable it's going to waste a lot more battery per press if it's spending that long trying to establish a connection to Amazon.  I'll keep an eye on it though, cracked open the old dead one and stuck a new AAA battery in it so it can act as my guinea pig to see if Amazon will kill it within the next few weeks.

edit: Also occurs to me that MAYBE by ensuring that no account is linked with the button and instead using the wav file exploit to write wifi data to it, you might be immune to getting your firmware messed with, since the Dash might try to authenticate with Amazon's servers and fail at an earlier stage if no account is associated with it.  Though that's just a total guess with no actual idea about how it all works.  Well, I'll keep an eye on it for a while and see what happens.  So far no problems after pressing it multiple times per day with internet connection enabled from 4-28 to 9-14, though I'll try to come back and edit this if it ever does get remote bricked going forward.

**Comment by quantumfrost** [comment]
Hey @OrangeFlavored, how did it turn out? Did your device get bricked, and how long did the battery last after letting it connect?

**Comment by OrangeFlavored** [comment]
Nah, no bricks and I used a few dash buttons multiple times a day up until a few months ago when I phased them out in favor of an Aqara Cube which I now use for pretty much the same stuff I used the dash buttons for.  Battery life did seem to consistently be better with the connection enabled which makes sense since they didn't spend as long trying to connect out and shut back off quickly.

**Comment by quantumfrost** [comment]
Awesome, thanks for the reply! Aqara Cube seems pretty cool too.
